### ISSUE TITLE
Discussion: Should we make reward function dependent on current score?

### ISSUE THREAD

**Issue by patrickvonplaten** [patrickvonplaten/TRexGameRL#22]

**Comment by kilsenp** [comment]
This basically would mean that we are saying that the current prediction depends on previous behavior (not just last 4 frames). Right now I would say this doesn't make sense, but what's your thinking behind that?

**Comment by patrickvonplaten** [comment]
In this post they did it: 
https://medium.com/acing-ai/how-i-build-an-ai-to-play-dino-run-e37f37bdf153

True that the markov property would be kind of broken then - let's try first without

**Comment by kilsenp** [comment]
"I trained my model for around 2 million frames **for a week**. 1st million steps were used for fine tuning the game parameters and fixing bugs. The last million training frames showed improvement in game scores reaching a maximum score of **265** till now. We can observe that loss has stabilized for the last million steps and stays low with minute fluctuations." This is a lot worse then the vdutor implementation where he reaches a score of **1600** after **half a day** of training. So we should take ideas from there with a grain of salt, but maybe let's look at some deep mind papers for the atari games.

**Comment by patrickvonplaten** [comment]
Ok, I guess we can just try out both possibilities. 
When u trained the model - what configs did u use?
- loss function? 
- exploration decay (epsilon)?
- optimizer?

**Comment by patrickvonplaten** [comment]
And 2 or 3 actions?

**Comment by kilsenp** [comment]
yeah true, I had trained for around 400 rounds with Adam and MSE, using the same epsilon strategy as vdutor, exponential decay. I had changed the espilon_decay to 0.97 because at 0.997 you are still pretty much random at 400 epochs. But I didn't get any noteworthy results by that point, that I wanted to try 0.997 some other time.
What I did note though, that it makes sense to put in a pause after the game has just started, where the frames are still empty. Otherwise memory will be full with those. I had set that pause to 3.5s.

**Comment by patrickvonplaten** [comment]
reward function is a hyperparameter we have to tune. We should prob try out:
- 2 vs 3 actions 
- different fixed values for actions 
- reward function dependent on other stuff
